platform/src/lib.rs

```diff
@@ -0,0 +1,20 @@
+// Licensed under the Apache-2.0 license
+
+pub use default::*;
+mod default;
+
+const MAX_CERT_SIZE: usize = 2048;
+
+pub enum PlatformError {
+    CertificateChainError,
+    NotImplemented,
+}
+
+pub struct CertChunk {
+    pub bytes: [u8; MAX_CERT_SIZE],
+    pub len: usize,
+}
+
+pub trait Platform {
+    fn get_certificate_chain(offset: u32, size: u32) -> Result<CertChunk, PlatformError>;
```

Comment:
I think returning CertChunk here requires an unnecessary copy. Instead, could do

```rust
/// Copies the out.len() bytes of the certificate chain starting at offset.
fn get_certficiate_chain(offset: u32, out: &mut [u8]) -> Result<(), PlatformError>
```

Reply:
The caller is just asking for as much of it as they want. `get_certificate_chain` would fill the whole `out` buffer.

One thing we could add is to return the number of bytes written. This way if the caller asks for more than are left, `get_certificate_chain` will just return the smaller number so the caller knows.